### DISCUSSION TITLE
Regarding some questions about DII

### DISCUSSION THREAD

**Discussion by huasheng12121** [jamesjiadazhan/dietaryindex#224] [Q&A] [answered]
Thank you to the author for providing such an excellent package for everyone to use. 

  I have some questions regarding the calculation of the Dietary Inflammatory Index (DII) that I would like to discuss with both the community and the author. 

 Q1: I noticed that in the calculation process of DII, attention needs to be paid to the units of nutrients. In the DII_NHANES_FPED function, CAFFEINE is processed as follows: DR2TCAFF/1000, and VITD is processed as follows: if (has_DR1TVD): DR1TVD * 0.025. Considering the units in the DII function and the unit representations in the NHANES database, I can understand the unit change for DR2TCAFF, but for DR1TVD, the unit in NHANES matches the required unit for VITD in both the DII function and the DII_NHANES_FPED function. Why is it necessary to multiply it by 0.025 to convert it to IU units?

Q2: I've noticed that in the DII_NHANES_FPED function, N3FAT is defined as DR1TP183 + DR1TP184 + DR1TP205 + DR1TP225 + DR1TP226, and N6FAT is defined as DR1TP183 + DR1TP204. However, in some literature, I've seen authors categorize DR1TP183 + DR1TP184 + DR1TP205 + DR1TP225 + DR1TP226 as N3FAT and DR1TP182 + DR1TP204 as N6FAT. Could categorizing DR1TP183 both as N3FAT and N6FAT in this DII_NHANES_FPED function be considered inappropriate?

Q3: If I want to use the DII function to calculate the Dietary Inflammatory Index for my own data, how should I calculate nutrients for two days? Is it sufficient to just modify REPEATNUM = 2? With this code: ALCOHOL_DII= DR1TALCO,  ALCOHOL_DII = DR2TALCO? Including this parameter in the function causes an error, stating that ALCOHOL_DII has multiple corresponding parameters. How should this code be written?

**Comment by jamesjiadazhan** [comment, marked as answer]
Thanks a lot for your detailed feedback.

You are right. The vitamin D does not need unit conversion in NHANES. Also, the DR1TP183 should not be treated as N6FAT. Instead, DR1TP182 should be used. I have updated these parts.

And yes, just modify REPEATNUM = 2. Please refer to the following example. You should use $ to specify the variable for each parameter.
#' @examples
#' data("DHQ3_exp")
#' DII(DHQ3_exp, DHQ3_exp$`Respondent ID`, 1, DHQ3_exp$`Alcohol (g)`, DHQ3_exp$`Vitamin B12 (mcg)`, DHQ3_exp$`Vitamin B6 (mg)`)

**Reply by huasheng12121** [reply]
Thank you for considering my feedback. I have one more request: could you provide references for the international mean values used in the calculation process of the DII_NHANES_FPED function? Specifically, I am looking for the sources of Overall_inflammatory_score, Global_mean, and SD in the function code. I have searched extensively but have been unable to find the origins of these values.

**Reply by huasheng12121** [reply]
I would like to examine other articles or sources that detail the general calculation process for international mean values, but I have been unable to find them. Thank you.

**Reply by jamesjiadazhan** [reply]
I included all references in the manual page: https://github.com/jamesjiadazhan/dietaryindex/blob/main/Manual.md

**Reply by huasheng12121** [reply]
Thanks for the link, I found the literature I needed.
